### PR TITLE
Adding complete tests for wrapping in pagination and export

### LINKED ISSUE #111: Add format/wrapping unit tests for export_as_txt and pagination
just a follow-up of #109

### PR DESCRIPTION
**Pull request by Melkiades**
Fixes #111 and #112

### CODE REVIEW

**Comment by gmbecker** [comment]
> Fixes #111 and #112

This PR does not appear to fix either of these bugs, because it only changes tests...

**Comment by Melkiades** [comment]
> > Fixes #111 and #112
> 
> This PR does not appear to fix #112 (and related #114), because it only changes tests...

Still working on it! sorry, it should have been a draft atm

File: tests/testthat/test-export.R [hunk 1]
```diff
@@ -20,3 +20,16 @@ testthat::test_that("Listing print correctly, with paginate", {
...
+  dat <- ex_adae
+  lsting <- as_listing(dat[1:10, ],
+    key_cols = c("USUBJID", "AESOC"),
+    disp_cols = c("USUBJID", "AESOC")
+  )
+
+  main_title(lsting) <- "this is some title"
+  main_footer(lsting) <- "this is some footer"
+
+  cat(export_as_txt(lsting, colwidths = c(10, 10), paginate = TRUE, lpp = 15))
```
Comment (Melkiades) [inline tests/testthat/test-export.R:34]: @ayogasekaram I would work on this to fix the export

**Comment by Melkiades** [comment]
@gmbecker why alias is necessary?

**Comment by gmbecker** [comment]
> @gmbecker why alias is necessary?
I'm not completely sure what you're asking here, but if it is waht I think you might be saying, its to prevent what I described here: https://github.com/insightsengineering/rlistings/issues/112#issuecomment-1506111497

The problem is that unlike `TableTree` objects in `rtables`, `listing_df`s inherit from a class that the default `toString` method (thinks it) can process.  So we need to prevent that.

File: R/rlistings_methods.R [hunk 2]
```diff
@@ -16,14 +16,21 @@
...
 #' @export
 #' @param x listing_df. The listing.
-#' @param ... dots. Unused
+#' @param ... dots. See `toString` method in \code{formatters} for list
+#' of parameters.
 #' @method print listing_df
 #' @name listing_methods
 print.listing_df <- function(x, ...) {
-  cat(toString(matrix_form(x)))
+  cat(toString(matrix_form(x), ...))
```
Comment (Melkiades) [inline R/rlistings_methods.R:24]: I think this is better, so it is possible to use colwidths also from print

File: R/rlistings_methods.R [hunk 3]
```diff
@@ -16,14 +16,21 @@
 #'
 #' @export
 #' @param x listing_df. The listing.
-#' @param ... dots. Unused
+#' @param ... dots. See `toString` method in \code{formatters} for list
+#' of parameters.
 #' @method print listing_df
 #' @name listing_methods
 print.listing_df <- function(x, ...) {
```
Comment (Melkiades) [inline R/rlistings_methods.R:23]: spell out relevant parameters

Comment (Melkiades) [reply R/rlistings_methods.R:23]: remember parameter description from formatters

**Comment by Melkiades** [comment]
It is complete now @shajoezhu @gmbecker

**Review by shajoezhu — APPROVED** [review]
Brilliant! Thanks @Melkiades

**Pull request merged**